extensions/wakatime/package.json

```diff
@@ -16,47 +16,66 @@
       "subtitle": "WakaTime",
       "title": "Show Summary",
       "description": "Show your WakaTime Summary",
-      "mode": "view"
+      "mode": "view",
+      "preferences": [
+        {
+          "name": "apiKey",
+          "description": "Use your WakaTime API key to use your account",
+          "type": "password",
+          "required": true,
+          "title": "WakaTime API Key"
+        }
+      ]
     },
     {
       "name": "leaderboard",
       "subtitle": "WakaTime",
       "title": "Show Public Leaderboard",
-      "description": "Show your rank in the public global leaderboard",
+      "description": "An overview of the public leaderboard. Check your rank against others all around the world.",
       "mode": "view"
     },
     {
       "name": "private-leaderboards",
       "subtitle": "WakaTime",
       "title": "Show Private Leaderboards",
       "description": "Show your private leaderboards",
-      "mode": "view"
+      "mode": "view",
+      "preferences": [
+        {
+          "name": "apiKey",
+          "description": "Use your WakaTime API key to use your account",
+          "type": "password",
+          "required": true,
+          "title": "WakaTime API Key"
+        }
+      ]
     }
   ],
   "preferences": [
     {
       "name": "apiKey",
-      "required": true,
+      "required": false,
       "title": "WakaTime API Key",
```

Comment:
Since I made this optional, I had to repeat it in the `summary` and `private-leaderboards` commands that require it.

I don't know if there's a better way of handling this, because I expected the `apiKey` preference to be shared across all commands. But adding it to one command doesn't apply it to the others/entire extension.

Reply:
Unfortunately there's no current way of defining a per-extension required preference and make it not required for a particular command, so I guess this is the only way to go about it currently. 😅

Reply:
This works! My only problem is the user still has to provide the API key for that command as well. It's still better than the way I had it before, because now it's only one other command that asks for it.

Just to be sure, is there a way to set the preference programmatically? I checked but didn't find anything.

Reply:
> the user still has to provide the API key for that command as well

do you mean the fact that the field still appears in preferences for that particular command? or something else?

The current version of the branch works as follows for me:

https://user-images.githubusercontent.com/1155589/198271493-d3f58d8c-3c3e-422e-9ffc-d0cf4ce71f3c.mov

Reply:
That's not my direct problem with it, but yeah I'll like for the same API key to be used across the extension and updating in the public leaderboard won't update the extension-wide one.